### PR TITLE
DX12 dependencies fix

### PR DESCRIPTION
**Pull request by Nekotekina**

### CODE REVIEW

**Comment by danilaml** [comment]
Doesn't build on win10 for me.

**Comment by danilaml** [comment]
image

**Comment by Zangetsu38** [comment]
hmm I was going to make PR :3 https://github.com/Zangetsu38/rpcs3/commit/ddccc0c6fdc96eb51ee31081bd3e265f4dcec4ad

**Comment by danilaml** [comment]
Anyway, what is the real point of this PR?
As I understood @vlj PR intentions, he just wanted to better separate d3d12 build and make rpcs3 load it dynamically (so it'd work on win8) (also make it a little harder to certain bad persons to break dx12).

**Comment by Zangetsu38** [comment]
@danilaml Neko does not like Windows 10 :3

**Comment by vlj** [comment]
Since cmake 3.4 is now released, couldn't we use cmake as our build system instead of VS anyway ? It's possible to group file by filters. The only drawback is that precompiled header are not well supported.

**Comment by Nekotekina** [comment]
What about now?

**Comment by Bigpet** [comment]
@vlj well we're using cotire for cmake already, I would hope that it could at the very least also provide precompiled headers for VS.

But I don't think CMake supports debug and release configurations in one solution very well, so that would be kind of annoying for Windows developers.

**Comment by Nekotekina** [comment]
@danilaml Those who compile with Win10 SDK may break the compilation for 8.1, so why not just fix the platform for everyone? It should work anyway, and 8.1 should be always available.

**Comment by danilaml** [comment]
@Nekoteki will only be able to test in 5 hours.  How would it break compilation on win8(it usually the other way around because appveyor didn't build dx12).10 sdk is available on win7-10. By that we should use win7 sdk.

File: asmjitsrc/asmjit.vcxproj [hunk 1]
```diff
@@ -68,6 +68,7 @@
   <PropertyGroup Label="Globals">
     <ProjectGuid>{AC40FF01-426E-4838-A317-66354CEFAE88}</ProjectGuid>
     <RootNamespace>asmjit</RootNamespace>
+    <WindowsTargetPlatformVersion>8.1</WindowsTargetPlatformVersion>
```
Comment (Zangetsu38) [inline asmjitsrc/asmjit.vcxproj]: Not set TargetPlatform

File: rpcs3/D3D12GSRender.vcxproj [hunk 2]
```diff
@@ -13,7 +13,7 @@
   <PropertyGroup Label="Globals">
     <ProjectGuid>{FAC9B17B-F4B8-4B75-8AEB-C8C7CB92B078}</ProjectGuid>
     <RootNamespace>D3D12GSRender</RootNamespace>
-    <WindowsTargetPlatformVersion>10.0.10240.0</WindowsTargetPlatformVersion>
+    <WindowsTargetPlatformVersion>8.1</WindowsTargetPlatformVersion>
```
Comment (Zangetsu38) [inline rpcs3/D3D12GSRender.vcxproj]: Same, only just remove TargetPlatform

**Comment by danilaml** [comment]
@Nekotekina Still doesn't compile (with slightly less errors 3133 -> 3128)

**Comment by Nekotekina** [comment]
What errors? Can you try plain build (Release)?

**Comment by danilaml** [comment]
@Nekotekina same as before (I'm trying Release + LLVM).
The problem is the mixup of different SDK headers as usual.

**Comment by Nekotekina** [comment]
> Can you try plain build (Release)? Did you rebuild LLVM itself?

**Comment by danilaml** [comment]
@Nekotekina rebuiltsolution, now it works. Though I still can't see the point of this change.

File: rpcs3/emucore.vcxproj [hunk 3]
```diff
@@ -645,7 +645,7 @@
   <PropertyGroup Label="Globals">
     <ProjectGuid>{C4A10229-4712-4BD2-B63E-50D93C67A038}</ProjectGuid>
     <RootNamespace>emucore</RootNamespace>
-    <WindowsTargetPlatformVersion>10.0.10240.0</WindowsTargetPlatformVersion>
+    <WindowsTargetPlatformVersion>8.1</WindowsTargetPlatformVersion>
```
Comment (Zangetsu38) [inline rpcs3/emucore.vcxproj]: Same

File: rpcs3/rpcs3.vcxproj [hunk 4]
```diff
@@ -26,7 +26,7 @@
     <ProjectGuid>{70CD65B0-91D6-4FAE-9A7B-4AF55D0D1B12}</ProjectGuid>
     <Keyword>Win32Proj</Keyword>
     <RootNamespace>rpcs3</RootNamespace>
-    <WindowsTargetPlatformVersion>10.0.10240.0</WindowsTargetPlatformVersion>
+    <WindowsTargetPlatformVersion>8.1</WindowsTargetPlatformVersion>
```
Comment (Zangetsu38) [inline rpcs3/rpcs3.vcxproj]: Same, Just remove TargetPlatform, not set 8.1

**Comment by Nekotekina** [comment]
@Zangetsu38 I want to force platform 8.1, not to allow changing it.

**Comment by Zangetsu38** [comment]
@Nekotekina It is precisely not necessary, leave empty if you like me, it automatically uses 8.1.

**Comment by Nekotekina** [comment]
Added dynamic loading of d3dcompiler_47.dll, hope it works.

@Zangetsu38 You can use "Retarget Solution" functionality but I'm 99,9% sure you don't need.

**Comment by danilaml** [comment]
@Nekotekina  isn't d3dcompiler_47.dll loaded dynamically by default? It's DLL.

**Comment by vlj** [comment]
Actually the library is dll but the loader is statically linked in a stub .lib. So every time rpcs3 is started the loader is fired even if no d3d functionnality is requested.

**Comment by Zangetsu38** [comment]
@Nekotekina That's what I said, it is unnecessary to set Target 8.1, it obligatory set the default.

**Comment by danilaml** [comment]
@Zangetsu38 It's better to set it to be more explicit. This property was introduced by VS15 so all sln's without it use 8.1 sdk/target by default, if I'm not mistaken. (that was one of the problems with travis).

**Comment by vlj** [comment]
Does build works ? I fear that switching sdk version may trigger incompatibilities between projects. I think D3D12GSRender need 10240 sdk.

**Comment by Zangetsu38** [comment]
@vlj Build works

File: README.md [hunk 5]
```diff
@@ -21,7 +21,7 @@ If you want to contribute please take a took at the [Coding Style](https://githu
 __Windows__
 * [Visual Studio 2015](https://www.visualstudio.com/en-us/downloads/download-visual-studio-vs.aspx)
 * [Visual C++ Redistributable Packages for Visual Studio 2015](http://www.microsoft.com/en-us/download/details.aspx?id=48145)
-* [Cmake 3.1.0+](http://www.cmake.org/download/) (optional, required only for LLVM build; add to PATH)
+* [Cmake 3.4.0+](http://www.cmake.org/download/) (optional, required only for LLVM build; add to PATH)
```
Comment (danilaml) [inline README.md]: This is not true, btw

Comment (paulsapps) [inline README.md]: But it should/will be for cmake generated slns?

Comment (danilaml) [inline README.md]: It seems not.

**Pull request merged**